shared/js/content-scripts/block-cookie.js

```diff
@@ -0,0 +1,90 @@
+// Set up 3rd party cookie blocker
+(function cookieBlocker () {
+    function getFrameUrl () {
+        let url = document.location.href
+        let parentFrame = (document !== window.top) && window.parent
+        while (parentFrame && url && !url.startsWith('http')) {
+            try {
+                url = parentFrame.document.location.href
+            } catch (ex) {
+                // ignore 'Blocked a fr[ame with origin '...'
+                // from accessing a cro]ss-origin frame.' exceptions
+            }
+            parentFrame = (parentFrame !== window.top) && parentFrame.parent
+        }
+        return url
+    }
+
+    function clearInjectedCookiesAndBlock () {
+        // Clear previously set cookies
+        var cookies = document.cookie.split('; ')
+        for (var c = 0; c < cookies.length; c++) {
+            var d = window.location.hostname.split('.')
+            while (d.length > 0) {
+                var cookieBase = encodeURIComponent(cookies[c].split(';')[0].split('=')[0]) + '=; expires=Thu, 01-Jan-1970 00:00:01 GMT; domain=' + d.join('.') + ' ;path='
+                var p = location.pathname.split('/')
+                document.cookie = cookieBase + '/'
+                while (p.length > 0) {
+                    document.cookie = cookieBase + p.join('/')
+                    p.pop()
+                };
+                d.shift()
+            }
+        }
+
+        // disable setting cookies
+        document.__defineSetter__('cookie', function (value) { })
+    }
+
+    function enforceCookieExpiry () {
+        var cookieSetter = document.__lookupSetter__('cookie')
+        // Enforce 7 day max expiry
+        document.__defineSetter__('cookie', function (value) {
+            var today = new Date()
+            var nextWeek = new Date(today.getFullYear(), today.getMonth(), today.getDate() + 7)
+            var comps = value.split(';')
+            var expiryFound = false
```

Comment:
How about using `;max-age=604800`? I believe the browser will ignore 'expiries' if 'max-age' is specified ([Source](https://mrcoles.com/blog/cookies-max-age-vs-expires/))